flyway_test.go

```diff
@@ -0,0 +1,140 @@
+	require.NoError(t, err)
+	return fmt.Sprintf("jdbc:postgresql:/%s:%s/%s", inspect.Name, defaultPostgresPort, defaultPostgresDbName)
+}
+
+func (c *intPostgresContainer) getExternalUrl(t testing.TB, ctx context.Context) string {
+	url, err := c.ConnectionString(ctx)
+	require.NoError(t, err)
+	return fmt.Sprintf("%ssslmode=disable", url) // disable ssl
+}
+
+func TestFlyway(t *testing.T) {
+	// given
+	ctx := context.Background()
+	networkContainer := createTestNetworkContainer(t, ctx)
+	postgresContainer := createTestPostgresContainer(t, ctx, networkContainer)
+
+	// when
+	flywayContainer, err := flyway.RunContainer(ctx,
+		flyway.WithNetwork(networkContainer.Name),
+		flyway.WithEnvUrl(postgresContainer.getInternalUrl(t, ctx)),
+		flyway.WithEnvUser(defaultPostgresDbUsername),
+		flyway.WithEnvPassword(defaultPostgresDbPassword),
+	)
+	require.NoError(t, err, "failed to run container")
+
+	// then
+	t.Cleanup(func() {
+		err := flywayContainer.Terminate(ctx)
+		require.NoError(t, err, "failed to terminate flyway container")
+
+		err = postgresContainer.Terminate(ctx)
+		require.NoError(t, err, "failed to terminate postgres container")
+	})
+
+	requireQuery(t, ctx, postgresContainer)
+
+	state, err := flywayContainer.State(ctx)
+	require.NoError(t, err, "failed to get container state")
+	require.Emptyf(t, state.Error, "failed to get container state")
+	require.Equal(t, 0, state.ExitCode, "container exit code was not as expected: migration failed")
+}
+
+func createTestNetworkContainer(t testing.TB, ctx context.Context) *testcontainers.DockerNetwork {
+	networkContainer, err := tcnetwork.New(ctx)
+	require.NoError(t, err)
+	require.NotNil(t, networkContainer)
+	return networkContainer
+}
+
+func createTestPostgresContainer(t testing.TB, ctx context.Context, networkContainer *testcontainers.DockerNetwork) *intPostgresContainer {
+	posgresContainerName := defaultPostgresContainerName
+	port := fmt.Sprintf("%s/tcp", defaultPostgresPort)
+
+	postgresContainer, err := tcpostgres.RunContainer(ctx,
+		withNetwork(posgresContainerName, networkContainer.Name),
+		testcontainers.WithImage(fmt.Sprintf("postgres:%s", defaultPostgresDbVersion)),
+		tcpostgres.WithDatabase(defaultPostgresDbName),
+		tcpostgres.WithUsername(defaultPostgresDbUsername),
+		tcpostgres.WithPassword(defaultPostgresDbPassword),
+		testcontainers.WithConfigModifier(func(config *container.Config) {
+			config.ExposedPorts = map[nat.Port]struct{}{
+				nat.Port(port): {},
+			}
+		}),
+		testcontainers.WithWaitStrategy(
+			wait.ForLog("database system is ready to accept connections").
+				WithOccurrence(2).
+				WithStartupTimeout(10*time.Second)),
+	)
+	require.NoError(t, err)
+
+	return &intPostgresContainer{
+		postgresContainer,
+	}
+}
+
+func withNetwork(name, network string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) error {
+		req.Name = name
+		req.Networks = []string{network}
+		return nil
+	}
+}
```

Comment:
You can use the existing functional option instead of this one.

Reply:
obviously I'll get you to re-review this area, but just want to point out I'm fishing a bit here :)